tests/Integration/Behaviour/Features/Context/Domain/Product/RelatedProductsFeatureContext.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain\Product;
+
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\RemoveAllRelatedProductsCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\SetRelatedProductsCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Query\GetRelatedProducts;
+use PrestaShop\PrestaShop\Core\Domain\Product\QueryResult\RelatedProduct;
+
+class RelatedProductsFeatureContext extends AbstractProductFeatureContext
+{
+    /**
+     * @Then product :productReference should have no related products
+     *
+     * @param string $productReference
+     */
+    public function assertProductHasNoRelatedProducts(string $productReference): void
+    {
+        $productId = $this->getSharedStorage()->get($productReference);
+        try {
+            $relatedProducts = $this->getQueryBus()->handle(new GetRelatedProducts($productId));
+            Assert::assertEmpty(
+                $relatedProducts,
+                sprintf('Product %s expected to have no related products', $productReference)
+            );
+        } catch (DomainException $e) {
```

Comment:
You shouldn't catch such a wide exception If something unexpected happens because some code was modified it could hide the problem You should only catch the exception you expect to test
And if you have no error assertions to make after using this command it even should be left with a try/catch

It's better to let it throw the exception and crash so we can see right away what is the problem